pyo3-ffi/src/ceval.rs

```diff
@@ -73,8 +73,12 @@ extern "C" {
     fn _Py_CheckRecursiveCall(_where: *mut c_char) -> c_int;
 }
 
-// skipped Py_EnterRecursiveCall
-// skipped Py_LeaveRecursiveCall
+extern "C" {
+    #[cfg(any(Py_3_9, not(Py_LIMITED_API)))]
+    pub fn Py_EnterRecursiveCall(arg1: *const c_char);
+    #[cfg(any(Py_3_9, not(Py_LIMITED_API)))]
+    pub fn Py_LeaveRecursiveCall();
```

Comment:
It looks like these were added to the stable API in 3.9, and also PyPy has them:

```suggestion
    #[cfg(Py_3_9)]
    #[cfg_attr(PyPy, link_name = "PyPy_EnterRecursiveCall")]
    pub fn Py_EnterRecursiveCall(arg1: *const c_char);
    #[cfg(Py_3_9)]
    #[cfg_attr(PyPy, link_name = "PyPy_LeaveRecursiveCall")]
    pub fn Py_LeaveRecursiveCall();
```

(In older versions they were implemented [as macros using unlimited API features](https://github.com/python/cpython/blob/f78733b332966cef7d0c61165389549f4ffe0727/Include/ceval.h#L89-L91), so we could maybe support in the unlimited API in older versions if @samuelcolvin needs that?)